### ISSUE TITLE
Suggestion: Add option to not cache back/forward buttons on browser

### ISSUE THREAD

**Issue by robgraeber** [dieulot/instantclick#133]
Depending on the website, it might not make sense to cache the back/forward button. (e.g. if you had an ecommerce site and you made changes in your shopping cart then went back.)

I got a basic implementation working, I'd be happy to submit a PR if your interested.

My implentation for reference: https://github.com/robgraeber/instantclick2/commit/b0c0443c54101def72dd78aac393c26e71d26821

Ideally instantclick could also preload the back/forward buttons when the mouse leaves the browser window.

**Comment by robgraeber** [comment]
I thought about this some more and it might not make sense, managing the scroll position while trying to fetch a page via the back btn is a big fan of worms.

**Comment by dieulot** [comment]
Yes, that might be useful.

I know [Turbolinks](https://github.com/turbolinks/turbolinks) is fetching pages when the user is going back when the page isn't in Turbolinks's cache (I guess they don’t handle scrolling though).

What did you mean by ”managing the scroll position”? Do you remember what scrolling problems you encountered?

**Comment by dieulot** [comment]
Yes, that may be useful in some cases. Though ideally you would just update elements on `restore` so that going back is still instant. A better way might be to add a `data-instant-keep-on-restore` attribute that would handle that: saving those elements before changing the page, and applying them to the page after changing it (both in `changePage()`).

I know [Turbolinks](https://github.com/turbolinks/turbolinks) is fetching pages when the user is going back when the page isn't in Turbolinks's cache (I guess they don’t handle scrolling though).

What did you mean by ”managing the scroll position”? Do you remember what scrolling problems you encountered?
